### ISSUE TITLE
[CheckOutlinesUFO.py] KeyError: missing glyph that, well, is correctly missing

### ISSUE THREAD

**Issue by robinsdg** [adobe-type-tools/afdko#228]
Running `checkOutlinesUFO` on a specific file throws errors about missing glyphs, and this is sensible, but is it the right behaviour? If something is defined in `lib.plist` but missing actual data, should it not skip/delete the reference in the flattening process?

```
$ checkOutlinesUFO -e Regular/font-flat.ufo/
. 
A There is an overlap.
Traceback (most recent call last):
  File "/Users/robin.mientjes/bin/FDK/Tools/osx/../SharedData/FDKScripts/CheckOutlinesUFO.py", line 1004, in <module>
    run(sys.argv[1:])
  File "/Users/robin.mientjes/bin/FDK/Tools/osx/../SharedData/FDKScripts/CheckOutlinesUFO.py", line 932, in run
    skip = fontFile.checkSkipGlyph(glyphName, options.checkAll)
  File "/Users/robin.mientjes/bin/FDK/Tools/osx/../SharedData/FDKScripts/CheckOutlinesUFO.py", line 217, in checkSkipGlyph
    width, outlineXML, skip = self.ufoFontHashData.getOrSkipGlyphXML(glyphName, doAll)
  File "/Users/robin.mientjes/Code/AFDKO/Tools/SharedData/FDKScripts/ufoTools.py", line 689, in getOrSkipGlyphXML
    glyphFileName = self.glyphMap[glyphName]
KeyError: 'A.smcp'
```

**Comment by robinsdg** [comment]
This is perhaps not a significant problem for anyone else, I recognise that. I just like to work inside of the `makeotf` process structure, so that I am more readily prepared for production exports. It also helps me to organise my files better. So a few empty spots where `lib.plist` says there should be glyphs, but that obviously are missing, should maybe not throw an error but rather just remove the reference from the flattened font.

**Comment by cjchapman** [comment]
I agree that this scenario should not cause the code to crash like that.

I just committed a change yesterday that I think will prevent this from happening in the scenario you described, though it will still happen if you specify such a glyph on the command line. The PR (which **is** merged to the master branch) is here:
https://github.com/adobe-type-tools/afdko/pull/227/files

I've got a proposed fix for the case in which you specify such a glyph on the command line in PR 229:
https://github.com/adobe-type-tools/afdko/pull/229/files
(this one is **not** merged to the master branch yet, as it is awaiting review)

**Comment by miguelsousa** [comment]
Fixed with #229
